Make/Makefile.incl

```diff
@@ -0,0 +1,99 @@
+
+# Generic Makefile to be included to support building code using the following examples:
+#
+# # Build an executable from all .cpp files in the current directory (recursively)
+# # with includes from ../other and library flags -L ../other -lother -- aka ../other/libother.a
+#
+# all:
+# TOPDIR=../..
+# EXE=myprog
+# CXXFLAGS+= -I ../other/
+# LIBS+=-L ../other -lother # links to libother.a
+# include $(TOPDIR)/Make/Makefile.incl
+
+# Default target
+all:
+
+# Compliler info
+CXX=g++
+CXXFLAGS+= -std=c++17
+CXXFLAGS+= -Wall
+CXXFLAGS+= -Werror
+CXXFLAGS+= -O3
+
+# Also generate dependency files when compiling objects
+CXXFLAGS_MAKE_DEPEND=-MMD
+
+# Archiver Info
+AR=ar
+ARFLAGS+=rcs
+
+# File definitions
```

Comment:
Do all this dynamically.  Doesn't work from Windows, sorry to say ;-)  We'll deal with that later.